src/state_machine.rs

```diff
@@ -322,6 +351,36 @@ impl StateMachine {
         }
     }
 
+    /// Query for a result, or yield: Err(NothingAvailable), Err(Disconnected).
+    #[cfg(feature = "use-mock-crust")]
+    pub fn try_step(&mut self, outbox: &mut EventBox) -> Result<(), TryRecvError> {
```

Comment:
Given a seed is it not guaranteed this function will be called exactly `n` times ? If true why would there be non-determinism ? So say each time you enter the function you generate a something random based on which you will insert the timer actions in the array. Why would this be non-deterministic if the global seed is known ?